opennms-base-assembly/src/main/resources/bin/_lib.sh

```diff
@@ -47,3 +47,45 @@ __onms_read_conf() {
     rm "${__onms_read_conf_tmp_file}"
   fi
 }
+
+__onms_is_absolute() {
+  case "$1" in
+    [/]*)
+      return 0
+      ;;
+    *)
+      ;;
+  esac
+  return 1
+}
+
+__onms_get_absolute_path() {
+  __abspath_dir="$(dirname "$1")"
+  __basename="$(basename "$1")"
+  pushd "$__abspath_dir" >/dev/null 2>&1 || exit 1
```

Comment:
Couldn't we just do the following:
`echo "${__abspath_dir}/${__basename}"`

Reply:
If you call `__onms_get_absolute_path foo` that will return `.` as dirname, and `foo` as basename.  So just echoing would print `./foo` rather than getting the full path of `.`.  This is a trick to ensure relative paths get turned into absolute.